src/transformers/modeling_outputs.py

```diff
@@ -227,6 +227,55 @@ class BaseModelOutputWithPastAndCrossAttentions(ModelOutput):
     cross_attentions: Optional[Tuple[torch.FloatTensor]] = None
 
 
+@dataclass
+class BaseModelOutputWithPoolingAndPastAndCrossAttentions(ModelOutput):
```

Comment:
This is starting to be a mouthful! Do we have a better name for this? I don't know what it's used for so can't offer any suggestions :-/

Reply:
It should also be documented!

Reply:
I agree - it's getting a bit too long. Logically, all model outputs that have `CrossAttention` should also have a cache and if after this PR all `BaseModelOutputWithPoolingAndCrossAttentions` would be replaced by `BaseModelOutputWithPoolingAndPastAndCrossAttentions` => so my proposition would be to just keep `BaseModelOutputWithPoolingAndCrossAttentions` and add `past_key_values` to it. Would that be ok?

Reply:
Same thing with `CausalLMOutputWithCrossAttentions` -> it would nowhere be used anymore after this PR. I'd also propose here to delete `CausalLMOutputWithPastAndCrossAttentions` (which is only used for GPT2) and replace it all by  `CausalLMOutputWithCrossAttentions`. Is that ok in terms of breaking backward compatibility?

Reply:
The act of renaming the outputs is not breaking (at least not sufficiently for us to care) but the act of adding something to the output is breaking if it's not None in a default/common behavior. So is the new field `None` for old uses?

Reply:
@sgugger `past_key_values` will be `None` when the model is used as `encoder` or when `use_cache` is `False` (which is the default behavior)

Reply:
If we follow my approach:
- For `CausalLMOutputWithCrossAttentions` this would mean -> no breaking changes for `GPT2` as all indexing and output length stay the same, only the class name is changed from `CausalLMOutputWithPastAndCrossAttentions` to `CausalLMOutputWithCrossAttentions`. For all other models *a.k.a* `RobertaForCausalLM`, `BertForCausalLM` and `BertGenerationDecoder` -> the output class is changed and the default behavior is to now have an additional `past_key_values` output by default (`use_cache=True` by default in those classes). So the difference from now to my approach is that in my approach the **type** of the output class stays the same while currently the type is changed. Also note that these models are only really used in EncoderDecoder settings (there are no pre-trained weights for standalone Bert models) so that those models are expected to have `past_key_values` as an output.
- For `BaseModelOutputWithPoolingAndCrossAttentions`, the default case stays the same. Only if `self.is_decoder=True` which is not the default case, will the `past_key_values` by part of the output. 

=> IMO, we can/should reuse the names `CausalLMOutputWithCrossAttentions` and `BaseModelOutputWithPoolingAndCrossAttentions`, **not** introduce `BaseModelOutputWithPoolingAndPastAndCrossAttentions` and delete `CausalLMOutputWithPastAndCrossAttentions`. It is more concise in terms of naming, IMO, it doesn't introduce more breaking changes as new names would have and it also forces us to directly implement the caching in the future.

Reply:
On a side-note, we should make a note in the top of the PR that the models `RobertaForCausalLM`, `BertForCausalLM`, `BertGenerationDecoder` and `EncoderDecoderModel` now - by default - also output the `past_key_values` during inference which is a small breaking change because of a new feature.

Reply:
I agree with @patrickvonplaten !